src/TracyHandler.php

```diff
@@ -36,21 +33,21 @@ public function __construct(
 	}
 
 
-	protected function write(array $record): void
+	protected function write(LogRecord $record): void
 	{
-		if (!isset($record['context']['exception']) || !$record['context']['exception'] instanceof Throwable) {
+		if (!isset($record->context['exception']) || !$record->context['exception'] instanceof Throwable) {
 			return;
 		}
 
-		if (!isset($record['context']['tracy_filename']) || !is_string($record['context']['tracy_filename'])) {
+		if (!isset($record->extra['tracy_filename']) || !is_string($record->extra['tracy_filename'])) {
 			return;
 		}
 
-		$this->lastMessage = $record['message'];
-		$this->lastContext = $record['context'];
+		$this->lastMessage = $record->message;
+		$this->lastContext = $record->context;
 
-		$exception = $record['context']['exception'];
-		$localName = $record['context']['tracy_filename'];
+		$exception = $record->context['exception'];
+		$localName = $record->extra['tracy_filename'];
```

Comment:
is the context -> extra indented?

I know context is readonly but can be modified using `with()`

E.g. sentry handler uses 'context' but [v3 support is not ready yet](https://github.com/getsentry/sentry-php/issues/1318) so it may be adapted there as well

https://github.com/getsentry/sentry-php/blob/6e8d3e24a7045b7222f8ad7987b7ebc6f0faf394/src/Monolog/Handler.php#L68

Reply:
Yes, it's intended. Extra should be more appropriate for processor data.

![image](https://user-images.githubusercontent.com/175109/169022910-3881b7c4-87d7-4ddd-a497-05f9fbc0b8c0.png)

Hope it does not break anything.